### PR TITLE
[REVIEW] Fix CategoricalDtype equality comparisons

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Fixes https://github.com/rapidsai/cudf/issues/5378

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/tests/test_categorical.py [hunk 1]
```diff
@@ -677,3 +679,26 @@ def test_add_categories_mixed_error():
...
+    # Categories are the same, but dtypes different
+    d1 = CategoricalDtype(
+        categories=np.array([1, 2, 3], dtype="int32"), ordered=False
+    )
+    d2 = CategoricalDtype(
+        categories=np.array([1, 2, 3], dtype="int64"), ordered=False
+    )
+
+    assert not d1 == d2
+
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_categorical.py:701]: Maybe also have a valid equality case?

Comment (brandon-b-miller) [reply python/cudf/cudf/tests/test_categorical.py:701]: I was thinking about that. But I'm not sure how to do that without constructing the exact same object twice, and then the situation is trivially equal.

Comment (galipremsagar) [reply python/cudf/cudf/tests/test_categorical.py:701]: We could do something like..
```python
>>> import pandas as pd
>>> x = pd.CategoricalDtype([1,2,3])
>>> y = pd.CategoricalDtype([1, 2, 3])
>>> x == y
True
>>> import cudf
>>> x = cudf.CategoricalDtype([1, 2, 3])
>>> y = cudf.CategoricalDtype([1, 2, 3])
>>> x == y
True
```

File: python/cudf/cudf/tests/test_categorical.py [hunk 2]
```diff
@@ -677,3 +679,28 @@ def test_add_categories_mixed_error():
 
     with pytest.raises(TypeError):
         gds.cat.add_categories(["a", "bd", "ef"])
+
+
+def test_categorical_equal_compare():
```
Comment (shwina) [inline python/cudf/cudf/tests/test_categorical.py:684]: Could this test be moved to `test_dtypes`? That's where the other tests are for `CategoricalDtype`, including the tests for equality suggested below

Comment (brandon-b-miller) [reply python/cudf/cudf/tests/test_categorical.py:684]: Yup - moving.

**Comment by brandon-b-miller** [comment]
ty all.

**Pull request merged**